go-controller/pkg/cni/helper_linux_test.go

```diff
@@ -0,0 +1,987 @@
+						},
+					},
+				},
+			},
+			errMatch: fmt.Errorf("failed to add pod route"),
+			netLinkOpsMockHelper: []ovntest.TestifyMockHelper{
+				{"LinkSetHardwareAddr", []string{"*mocks.Link", "net.HardwareAddr"}, []interface{}{nil}},
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil}},
+			},
+			cniPluginMockHelper: []ovntest.TestifyMockHelper{
+				{"AddRoute", []string{"*net.IPNet", "net.IP", "*mocks.Link"}, []interface{}{nil}},
+				{"AddRoute", []string{"*net.IPNet", "net.IP", "*mocks.Link"}, []interface{}{fmt.Errorf("mock error")}},
+			},
+		},
+		{
+			desc:    "test success path",
+			inpLink: mockLink,
+			inpPodIfaceInfo: &PodInterfaceInfo{
+				PodAnnotation: util.PodAnnotation{
+					IPs:      ovntest.MustParseIPNets("192.168.0.5/24"),
+					MAC:      ovntest.MustParseMAC("0A:58:FD:98:00:01"),
+					Gateways: ovntest.MustParseIPs("192.168.0.1"),
+					Routes: []util.PodRoute{
+						{
+							Dest:    ovntest.MustParseIPNet("192.168.1.0/24"),
+							NextHop: net.ParseIP("192.168.1.1"),
+						},
+					},
+				},
+			},
+			netLinkOpsMockHelper: []ovntest.TestifyMockHelper{
+				{"LinkSetHardwareAddr", []string{"*mocks.Link", "net.HardwareAddr"}, []interface{}{nil}},
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil}},
+			},
+			cniPluginMockHelper: []ovntest.TestifyMockHelper{
+				{"AddRoute", []string{"*net.IPNet", "net.IP", "*mocks.Link"}, []interface{}{nil}},
+				{"AddRoute", []string{"*net.IPNet", "net.IP", "*mocks.Link"}, []interface{}{nil}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			for _, item := range tc.netLinkOpsMockHelper {
+				call := mockNetLinkOps.On(item.OnCallMethodName)
+				for _, arg := range item.OnCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.RetArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			for _, item := range tc.linkMockHelper {
+				call := mockLink.On(item.OnCallMethodName)
+				for _, arg := range item.OnCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.RetArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			for _, item := range tc.cniPluginMockHelper {
+				call := mockCNIPlugin.On(item.OnCallMethodName)
+				for _, arg := range item.OnCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.RetArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			err := setupNetwork(tc.inpLink, tc.inpPodIfaceInfo)
+			t.Log(err)
+			if tc.errMatch != nil {
+				assert.Contains(t, err.Error(), tc.errMatch.Error())
+			} else {
+				assert.Nil(t, err)
+			}
+			mockNetLinkOps.AssertExpectations(t)
```

Comment:
L359: Same question as before. I think it is sufficient to check if `err != nil` and not the contents of the error itself.

Reply:
same response as above